tests/test_reweighting.py

```diff
@@ -0,0 +1,407 @@
+import numpy as onp
+import pymbar
+from jax import grad, jit
+from jax import numpy as np
+from jax import value_and_grad, vmap
+from jax.scipy.stats.norm import logpdf as norm_logpdf
+from scipy.stats import norm
+
+from timemachine.constants import BOLTZ
+from timemachine.datasets import fetch_freesolv
+from timemachine.fe.functional import construct_differentiable_interface_fast
+from timemachine.fe.reweighting import (
+    construct_endpoint_reweighting_estimator,
+    construct_mixture_reweighting_estimator,
+    interpret_as_mixture_potential,
+    one_sided_exp,
+)
+from timemachine.ff import Forcefield
+from timemachine.md.enhanced import get_solvent_phase_system
+
+# TODO: should gaussian test system be moved into a test fixture? or perhaps to the testsystems module?
+
+
+def make_gaussian_testsystem():
+    """normalized/unnormalized 1D Gaussian with a dependence on lambda and params"""
+
+    def annealed_gaussian_def(lam, params):
+        initial_mean, initial_log_sigma = 0.0, 0.0
+        target_mean, target_log_sigma = params
+
+        # lam = 0 -> (mean = 0, stddev = 1)
+        # lam = 1 -> (mean = target_mean, stddev = target_sigma)
+        mean = lam * target_mean - (1 - lam) * initial_mean
+        stddev = np.exp(lam * target_log_sigma + (1 - lam) * initial_log_sigma)
```

Comment:
Mostly for my own curiosity: any reason in particular to weight stddev in logspace?

Reply:
No reason, just wrote down an arbitrary 1D test system with the desired behavior at lam = 0 and lam = 1.

---
Aside: in the special case of annealing between two 1D gaussians with the same mean but different sigma, this is an optimal protocol, as mentioned in eqs. 70-71 of [[Gelmen, Meng, 1998]](http://www.stat.columbia.edu/~gelman/research/published/path2.pdf).

(However, it doesn't matter whether this test uses an efficient protocol, the test doesn't use the appropriate generalization for 1D gaussians with different mean, and the differences in sigma are small -- log space isn't required for stability.)